mapproxy/source/ogcapimaps.py

```diff
@@ -0,0 +1,189 @@
+    normalize_srs_code,
+    build_absolute_url,
+)
+from mapproxy.util.py import reraise_exception
+
+import logging
+
+log = logging.getLogger("mapproxy.source.ogcapimaps")
+
+# For testing
+reset_cache = False
+
+
+class OGCAPIMapsSource(WMSLikeSource):
+    def __init__(
+        self,
+        landingpage_url,
+        collection,
+        http_client,
+        coverage=None,
+        image_opts=None,
+        error_handler=None,
+        res_range=None,
+        transparent=None,
+        transparent_color=None,
+        transparent_color_tolerance=None,
+    ):
+        WMSLikeSource.__init__(
+            self,
+            image_opts=image_opts,
+            coverage=coverage,
+            res_range=res_range,
+            transparent_color=transparent_color,
+            transparent_color_tolerance=transparent_color_tolerance,
+            error_handler=error_handler,
+        )
+
+        self.landingpage_url = landingpage_url.rstrip("/")
+        self.collection = collection
+        self.http_client = http_client
+        if transparent is not None:
+            self.image_opts.transparent = transparent
+
+        self.lock = Lock()
+        # Below variables are protected under the lock
+        self._reset_caches()
+
+    def _reset_caches(self):
+        self.has_got_maps_list = False
+        self.map_format_to_href = {}
+        self.supported_srs = set()
+        self.map_crs_code_to_url = {}
+
+    def _build_url(self, href):
+        return build_absolute_url(self.landingpage_url, href)
+
+    def _get_maps_list(self):
+        with self.lock:
+            # Used by tests to avoid caching of "metadata"
+            global reset_cache
+            if reset_cache:
+                reset_cache = False
+                self._reset_caches()
+
+            if self.has_got_maps_list:
+                return
+            self.has_got_maps_list = True
+
+            headers = {"Accept": "application/json"}
+
+            url = self.landingpage_url
+            if self.collection:
+                url += "/collections/" + self.collection
+            try:
+                resp = self.http_client.open(url, headers=headers)
+            except HTTPClientError as e:
+                log.warning(f"Cannot retrieve {url}: %s", e)
+                reraise_exception(SourceError(e.args[0]), sys.exc_info())
+            try:
+                j = json.loads(resp.read().decode("utf-8"))
+            except Exception as e:
+                log.warning(f"Cannot parse response to {url} as JSON: %s", e)
+                reraise_exception(SourceError(e.args[0]), sys.exc_info())
+
+            if "links" not in j:
+                ex = SourceError(f"Could not retrieve 'links' in {url} response")
+                log.error(ex)
+                raise ex
+
+            for mimetype in ("image/png", "image/jpeg"):
+                href = find_href_in_links(
+                    j["links"],
+                    "http://www.opengis.net/def/rel/ogc/1.0/map",
```

Comment:
According to the standard, the value can also be `[ogc-rel:map]`: https://docs.ogc.org/is/20-058/20-058.html#req_collection-map_desc-links

Reply:
If no link with rel=http://www.opengis.net/def/rel/ogc/1.0/map or rel=[ogc-rel:map] is found, MapProxy should raise a NOTICE or WARNING that the specified collection does not contain an OGC API Maps endpoint.